src/main.rs

```diff
@@ -256,20 +263,149 @@ mod tests {
             Ok(res) => res,
             Err(_) => panic!("Could not parse IP Address"),
         };
-        let scanner = Scanner::new(addr, 1, 65535, 100, Duration::from_millis(10), true);
-        let scan_result = block_on(scanner.run());
+        let scanner = Scanner::new(addr, 1, 65535, 100, Duration::from_millis(100), true);
+        block_on(scanner.run());
         // if the scan fails, it wouldn't be able to assert_eq! as it panicked!
         assert_eq!(1, 1);
     }
+    #[test]
     fn does_it_run_ipv6() {
         // Makes sure te program still runs and doesn't panic
         let addr = match "::1".parse::<IpAddr>() {
             Ok(res) => res,
             Err(_) => panic!("Could not parse IP Address"),
         };
-        let scanner = Scanner::new(addr, 1, 65535, 100, Duration::from_millis(10), true);
-        let scan_result = block_on(scanner.run());
+        let scanner = Scanner::new(addr, 1, 65535, 100, Duration::from_millis(100), false);
+        block_on(scanner.run());
         // if the scan fails, it wouldn't be able to assert_eq! as it panicked!
         assert_eq!(1, 1);
     }
+    #[test]
+    fn does_it_run_quad_0() {
+        let addr = match "0.0.0.0".parse::<IpAddr>() {
+            Ok(res) => res,
+            Err(_) => panic!("Could not parse IP Address"),
+        };
+        let scanner = Scanner::new(addr, 1, 1000, 100, Duration::from_millis(500), true);
+        block_on(scanner.run());
+        assert_eq!(1, 1);
+    }
+    #[test]
+    fn zero_ports() {
+        let addr = match "0.0.0.0".parse::<IpAddr>() {
+            Ok(res) => res,
+            Err(_) => panic!("Could not parse IP Address"),
+        };
+        let scanner = Scanner::new(addr, 1, 1, 100, Duration::from_millis(50), true);
+        block_on(scanner.run());
+        assert_eq!(1, 1);
+    }
+    #[test]
+    fn backwards_ports() {
+        let addr = match "0.0.0.0".parse::<IpAddr>() {
+            Ok(res) => res,
+            Err(_) => panic!("Could not parse IP Address"),
+        };
+        let scanner = Scanner::new(addr, 10, 1, 100, Duration::from_millis(50), true);
+        block_on(scanner.run());
+        assert_eq!(1, 1);
+    }
+    #[test]
+    fn google_test() {
+        let addr = match "8.8.8.8".parse::<IpAddr>() {
+            Ok(res) => res,
+            Err(_) => panic!("Could not parse IP Address"),
+        };
+        let scanner = Scanner::new(addr, 400, 445, 100, Duration::from_millis(1500), true);
+        block_on(scanner.run());
+        assert_eq!(1, 1);
+    }
+    #[test]
+    fn test_mac_ulimit() {
```

Comment:
How do we know this test is being run in a mac environment?

Reply:
Travis CI runs all tests in OSX. This test tests for the specific problem of Mac's having a small ulimit, the test fails if it does not decrease the ulimit (as OSX will panic). 

See this as proof that we are testing on OSX:
https://github.com/RustScan/RustScan/blob/master/.travis.yml#L5

Reply:
Ohh, good one, thanks for explaining! 🎉 

Should we name these tests according to what they are trying to do and not the environment itself? For example `test_mac_ulimit` becomes `automatic_ulimit_reduction`

Reply:
Ah, sure! I'm a new compsci graduate and they don't teach you naming things :wink: but I'm learning :stuck_out_tongue: Give me a few minutes and I'll resubmit :)

Reply:
You will get the hang of it in no time 😄

Reply:
I'd like to just quickly share this document, :)

https://doc.rust-lang.org/1.0.0/style/